src/model_repository_manager.cc

```diff
@@ -2095,38 +2099,38 @@ ModelRepositoryManager::DependencyGraph::Swap(DependencyGraph& rhs)
   std::swap(global_map_ptr_, rhs.global_map_ptr_);
   nodes_.swap(rhs.nodes_);
   missing_nodes_.swap(rhs.missing_nodes_);
+  removed_nodes_.swap(rhs.removed_nodes_);
+}
+
+ModelRepositoryManager::DependencyNode*
+ModelRepositoryManager::DependencyGraph::GetNode(
+    const ModelIdentifier& model_id) const
+{
+  const auto it = removed_nodes_.find(model_id);
+  if (it != removed_nodes_.end()) {
+    return it->second.get();
+  }
+  auto* node = FindNode(model_id, false /* allow_fuzzy_matching */);
+  if (node == nullptr) {
+    throw std::invalid_argument("model_id '" + model_id.str() + "' not found");
```

Comment:
Out of curiosity, is there a good spot anywhere up the call chain to catch the exception and log an error? Or should this just crash the program?

Reply:
I just want to crash the program, because to recover from this failure, I think the model repository manager needs to be restarted, meaning unloading all the models and re-loading them again, which is not much difference from crashing the program and then restarting.

The good thing is, this should never happen in production under any circumstances, but only to make debugging easier if it happens (instead of debugging a `nullptr` dereference).